drainer/checkpoint/checkpoint.go

```diff
@@ -50,7 +50,7 @@ func NewCheckPoint(cfg *Config) (CheckPoint, error) {
 		err error
 	)
 	switch cfg.CheckpointType {
-	case "mysql", "tidb":
+	case "mysql", "tidb", "plugin":
```

Comment:
this means we must save checkpoint in a database for plugin type, how about adding a judgment for plugin: when `cfg.DB != nil` use `newMysql`, otherwise use NewFile